quic/s2n-quic-transport/src/endpoint/mod.rs

```diff
@@ -459,6 +459,17 @@ impl<Cfg: Config> Endpoint<Cfg> {
         ) {
             (packet, remaining)
         } else {
+            if Cfg::DcEndpoint::ENABLED
+                && endpoint_context.dc.on_possible_secret_control_packet(
+                    &dc::DatagramInfo::new(&remote_address),
+                    payload,
+                )
+            {
+                // This was a DC secret control packet, so we don't need to proceed
+                // with checking for a stateless reset
+                return;
+            }
```

Comment:
I'm surprised clippy doesn't complain about collapsing `if else`... but we should :)

Reply:
nvm, I got it